### PR TITLE
Updating Microsoft.Build.Sql for utilising in Visual Studio

### PR DESCRIPTION
**Pull request by Ri7Sh**

### CODE REVIEW

File: src/Microsoft.Build.Sql/sdk/Sdk.props [hunk 1]
```diff
@@ -26,10 +28,10 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">16.0</VisualStudioVersion>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">17.0</VisualStudioVersion>
     <!-- Default to the v16.0 targets path if the targets file for the current VS version is not found -->
```
Comment (ErikEJ) [inline src/Microsoft.Build.Sql/sdk/Sdk.props]: Update comment?

File: src/Microsoft.Build.Sql/sdk/Sdk.targets [hunk 2]
```diff
@@ -66,6 +72,7 @@
     <Folder Include="Properties" />
   </ItemGroup>
   
+
```
Comment (ErikEJ) [inline src/Microsoft.Build.Sql/sdk/Sdk.targets:69]: Why extra newline?

File: src/Microsoft.Build.Sql/sdk/Sdk.props (hunk 1 above)
Comment (Ri7Sh) [reply src/Microsoft.Build.Sql/sdk/Sdk.props]: @ErikEJ Looks like you are reviewing code from a single commit only. Comment is updated as well

File: src/Microsoft.Build.Sql/sdk/Sdk.props [hunk 3]
```diff
@@ -12,12 +12,17 @@
     <TargetFramework Condition="'$(TargetFramework)' == '' AND '$(NetCoreBuild)' == 'true'">netstandard2.1</TargetFramework>
   </PropertyGroup>
 
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
```
Comment (zijchen) [inline src/Microsoft.Build.Sql/sdk/Sdk.props:15]: Should this Import go to the very top?

File: src/Microsoft.Build.Sql/sdk/Sdk.props [hunk 4]
```diff
@@ -26,10 +31,16 @@
...
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">16.0</VisualStudioVersion>
-    <!-- Default to the v16.0 targets path if the targets file for the current VS version is not found -->
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">17.0</VisualStudioVersion>
+    <!-- Default to the v17.0 targets path if the targets file for the current VS version is not found -->
     <SSDTExists Condition="Exists('$(MSBuildExtensionsPath)\Microsoft\VisualStudio\v$(VisualStudioVersion)\SSDT\Microsoft.Data.Tools.Schema.SqlTasks.targets')">True</SSDTExists>
-    <VisualStudioVersion Condition="'$(SSDTExists)' == ''">16.0</VisualStudioVersion>
+    <VisualStudioVersion Condition="'$(SSDTExists)' == ''">17.0</VisualStudioVersion>
+  </PropertyGroup>
+
+  <!-- building in Visual Studio requires some sort of TargetFrameworkVersion. So we condition to NetCoreBuild as false to avoid failures -->
```
Comment (zijchen) [inline src/Microsoft.Build.Sql/sdk/Sdk.props:40]: nit: Move this block to near the top around line 14 so all the target framework defaults are together

File: src/Microsoft.Build.Sql/sdk/Sdk.targets [hunk 5]
```diff
@@ -39,9 +37,15 @@
...
   </PropertyGroup>
-
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   
+  <PropertyGroup>
+    <CSharpDesignTimeTargetsPath Condition="'$(CSharpDesignTimeTargetsPath)'==''">$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed\Microsoft.CSharp.DesignTime.targets</CSharpDesignTimeTargetsPath>
+  </PropertyGroup>
+
+  <Import Project="$(CSharpDesignTimeTargetsPath)" Condition="'$(CSharpDesignTimeTargetsPath)' != '' and Exists('$(CSharpDesignTimeTargetsPath)')" />
+  <Target Name="CoreCompile" />
```
[the remark this answers is not in the document]
Comment (Ri7Sh) [reply src/Microsoft.Build.Sql/sdk/Sdk.targets]: Added comment

File: src/Microsoft.Build.Sql/sdk/Sdk.props (hunk 3 above)
Comment (Ri7Sh) [reply src/Microsoft.Build.Sql/sdk/Sdk.props]: Moved the import to top.

File: src/Microsoft.Build.Sql/sdk/Sdk.props (hunk 1 above)
Comment (ErikEJ) [reply src/Microsoft.Build.Sql/sdk/Sdk.props]: Yeah, missed that

**Review by zijchen — APPROVED** [review]

File: src/Microsoft.Build.Sql/sdk/Sdk.targets [hunk 6]
```diff
@@ -6,30 +6,28 @@
...
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <DefineDebug>false</DefineDebug>
-    <DefineTrace>true</DefineTrace>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+  <PropertyGroup>
+    <WarningLevel Condition=" '$(WarningLevel)' == '' ">4</WarningLevel>
+    <TreatWarningsAsErrors Condition=" '$(TreatWarningsAsErrors)' == '' ">false</TreatWarningsAsErrors>
+    <BuildScriptName Condition=" '$(BuildScriptName)' == '' ">$(MSBuildProjectName).sql</BuildScriptName>
```
Comment (chlafreniere) [inline src/Microsoft.Build.Sql/sdk/Sdk.targets:12]: What is the `>` in BuildScriptName?

**Comment by czb182** [comment]
Should the latest version of VS be able to build this now, using latest  1.19-preview sdk?

I tried migrating one, but gives build error around .net framework.  I followed the migration docs.

Your project does not reference ".NETFramework,Version=v4.7.2" framework. Add a reference to ".NETFramework,Version=v4.7.2" in the "TargetFrameworks" property of your project file and then re-run NuGet restore.	Deloitte.Omnia.Materiality.Database	C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\NuGet\17.0\Microsoft.NuGet.targets	198

**Comment by dzsquared** [comment]
@czb182 unfortunately more than this PR is needed for support in Visual Studio. this work is ongoing!
you can track progress for SSDT on #180

**Pull request merged**